vertical-pod-autoscaler/docs/proposals/crash_loop_handling/README.md

```diff
@@ -0,0 +1,196 @@
+  - 0.95 for upper bound requests of CPU and Memory
+- For Target Estimator we only recommend Percentile Estimation with a safety margin of 15% by default
+- An exponential variation in the lower bound requests, upper bound requests which is determined by history of metrics monitored and pod life span. There is a "confidence" value on which the entire resource scaling is determined.
+- For lower bound and upper bound estimation a configurable safety margin (by default 15%) is applied
+- The current lower bound and upper bound estimation formula is as follows
+```
+scaledResource = originalResource * (1 + 1/confidence)^exponent
+
+confidence = (LastSampleTimeStart - FirstSampleTimeStart) / (no. of hours per day i.e 24)
+```
+
+Where,
+- **scaledResource**  : newly scaled integer value of the amount of the 
+                    resource  (CPU or Memory)
+- **originalResource**: current integer value of the amount of the
+                    resource (CPU or Memory)
+- **confidence**      : An non-negative integer number that heuristically measures 
+                    how much confidence the history aggregated in the AggregateContainerState provides. For a workload producing a steady stream of samples over N days at the rate of 1 sample per minute, this metric is equal to N. It is calcualted as the total count of samples and the time between the first and the last sample 
+- **exponent**        : A non-zero float exponent value which is determined by the 
+                    targetEstimator, lowedBoundEstimator or upperBoundEstimator. Each of these estimators are set through percentile estimations. targetEstimator is set to percentileEstimator while lowerBoundEstimator and upperBoundEstimator are set as confidence multipliers. For lowerBoundEstimator the exponent is set to -2 and for upperBoundEstimator it is set to +1
+
+## Proposal
+
+To address the concerns mentioned we are proposing this change in the recommendation engine of VPA recommender for autoscaler component.
+
+#### Scale Up:
+Doubling of resources based on current resource usage
+  1. Today we depend on the resource requests history from prometheus client for a history length of 8 days (default)
+  2. Instead, we could obtain current and past 10 minutes usage from prometheus client to get a nearest estimate of how much requests for resources is a pod making
+  3. In the event of a spike in resource, we can recommend the target estimator, requests as double the current resource usage. The lower bound and upper bound become the function of target estimator instead of directly recommending based on resource usage using percentile approach. Lower bound can be set to 0.5 times the target estimator value and upper bound can be set to 2 times the target estimator value.
+  4. This will ensure handling the spike as it is dependent on the current usage than a hysteresis of resource usage pattern and doubling will react exponentially to a potentially exponential spike in resouce usage.
```

Comment:
I'm afraid, that if the recommender just doubles the recommendation value, scale-up will happen way too fast and the real resource utilization of cluster nodes (e.g. Seed nodes) will decrease even further.

Reply:
That is one concern for sure and I did mention it in "Concerns" heading below. One way we are controlling this by using a parameter called `thresholdScaleUp` and `thresholdScaleDown`. So we scale only if we are going out of this threshold window. But again, it doesn't fully solve the thrashing problem but reduces the probability of it. I guess, I need to experiment more on this to give you a bit more numeric data representation to be honest.